### PR TITLE
Add Node.js logo

### PR DESCRIPTION
**Pull request by jasnell**
Refs: https://github.com/nodejs/node/issues/50668

### CODE REVIEW

File: main.jsx [hunk 1]
```diff
@@ -378,6 +378,12 @@ const PARTNER_LOGOS = [
     name: "Netlify",
     restrict: "horizontal",
   },
+  {
+    src: "/static/logos/nodejs.svg",
+    href: "https://nodejs.org",
+    name: "Node.js",
+    restrict: "horizontal",
```
Comment (jasnell) [inline main.jsx:385]:
```suggestion
    restrict: "vertical",
```

File: main.jsx [hunk 2]
```diff
@@ -378,6 +378,12 @@ const PARTNER_LOGOS = [
     name: "Netlify",
     restrict: "horizontal",
   },
+  {
+    src: "/static/logos/nodejs.svg",
+    href: "https://nodejs.org",
+    name: "Node.js",
+    restrict: "vertical",
```
Comment (jasnell) [inline main.jsx:385]:
```suggestion
    restrict: "horizontal",
```

**Comment by lucacasonato** [comment]
I'm not sure `restrict: "horizontal"` is actually implemented. I'll take a look.

**Comment by jasnell** [comment]
It's fine if it's not. I tried `restrict: 'vertical'` but it made the logo way too small relative to the others.

**Review by Ethan-Arrowood — APPROVED** [review]

**Pull request merged**